contrib_bots/lib/yoda_bot.py

```diff
@@ -0,0 +1,129 @@
+
+        original_content = message['content']
+
+        is_sentence = (original_content.startswith('@yoda'))
+
+        return is_sentence
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        stream = message['display_recipient']
+        subject = message['subject']
+
+        # this handles the message if its starts with @yoda
+        if original_content.startswith('@yoda'):
+            handle_input(client, original_content, stream, subject)
+
+handler_class = YodaSpeakHandler
+
+
+def send_to_yoda_api(sentence, api_key):
+    # function for sending sentence to api
+
+    try:
+        response = unirest.get("https://yoda.p.mashape.com/yoda?sentence=" + sentence,
+                               headers={
+                                   "X-Mashape-Key": api_key,
+                                   "Accept": "text/plain"
+                               }
+                               )
+
+        if response.code == 200:
+            return response.body
+        if response.code == 403:
+            raise ApiKeyError
+        else:
+            error_message = response.body['message']
+            logging.error(error_message)
+            error_code = response.code
+            error_message = error_message + 'Error code: ' + error_code + \
+                '. Did you follow the instructions in the `readme-yoda-bot.md` file?'
+            return error_message
+
+    except ssl.SSLError:
+        raise ServerError
+    except TypeError:
+        raise ServerError
+
+
+def format_input(original_content):
+    # replaces the '@yoda' with nothing, so that '@yoda' doesn't get sent to the api
+    message_content = original_content.replace('@yoda', '')
+    # gets rid of whitespace around the edges, so that they aren't a problem in the future
+    message_content = message_content.strip()
+    # replaces all spaces with '+' to be in the format the api requires
+    sentence = message_content.replace(' ', '+')
+    return sentence
+
+
+def handle_input(client, original_content, stream, subject):
+    try:
+        sentence = format_input(original_content)
+        reply_message = send_to_yoda_api(sentence, get_api_key())
+        send_message(client, reply_message, stream, subject)
+
+    except ServerError:
+        reply_message = 'The service is temporarily unavailable, please try again.'
+        logging.error(reply_message)
+        send_message(client, reply_message, stream, subject)
+
+    except ApiKeyError:
+        reply_message = 'Invalid Api Key. Did you follow the instructions in the ' \
+                        '`readme-yoda-bot.md` file?'
+        logging.error(reply_message)
+        send_message(client, reply_message, stream, subject)
+
+
+def get_api_key():
+    # function for getting Mashape api key
+    home = os.path.expanduser('~')
+    api_key = open(home + '/yoda_api_key.txt').read().strip()
```

Comment:
You need to close this file as well. Better to use `with open` as it will cleanup the resources automatically.